js/piwik.js

```diff
@@ -499,11 +499,13 @@ if (typeof window.Matomo !== 'object') {
          */
         function beforeUnloadHandler() {
             var now;
-            isPageUnloading = true;
 
-            executePluginMethod('unload');
+            if (documentAlias.visibilityState === 'hidden') {
```

Comment:
Shouldn't that be more something like
```js
if (documentAlias.visibilityState !== 'hidden') {
    return;
}
isPageUnloading = true;
executePluginMethod('unload');
```

Otherwise blocking the UI below will always be done...